### DISCUSSION TITLE
Send to Kindle

### DISCUSSION THREAD

**Discussion by NovomirK** [petrovvlad/freeLib#67] [General]
Здравствуйте. Спасибо за программу. Всё работает. Не смог разобраться только с одним. При попытке отправить книгу непосредственно на аккаунт киндла программа вылетает. Система Kubuntu 20.04x64. Что-то я явно делаю не так. Ещё бы понять что. он конвертирует в mobi, всё хорошо, а потом: 
Error send e-mail.
munmap_chunk(): invalid pointer
Аварийный останов (стек памяти сброшен на диск)

Может быть подскажете решение?

**Comment by petrovvlad** [comment]
Проверил. У меня тоже не работает отправка почты. Буду искать причину.

**Reply by NovomirK** [reply]
Если что-то могу протестить, помочь и т.д. - пишите. С учётом, что я не очень продвинутый пользователь ))

**Comment by petrovvlad** [comment]
Спасибо за предложение помощи. Но в данном случае проблема в том, что для отправки почты используется сторонняя библиотека. Может потребоваться время для того, что бы разобраться с ней.

**Reply by NovomirK** [reply]
Тогда остаётся ждать новой версии. Отправлять книги можно и с почты. Ничего страшного.

**Comment by petrovvlad** [comment]
Отправка почты заработала. Пока не починил кодировку имени файла в письме.

**Reply by NovomirK** [reply]
Здорово! Спасибо большое. Уверен,  что и кодировка со временем починится. 
P.S. Я правильно понимаю, что там где адрес сервера нужно прописывать smtp.yandex.com , к примеру?
Жду релиз. На форуме Убунту про форк рассказал.

**Comment by petrovvlad** [comment]
Вроде бы поправил кодировку. Просьба проверить.

**Comment by NovomirK** [comment]
У меня всё работает. Спасибо огромное. Книга пришла и открылась. 
Может быть подскажете, а зачем он в названии добавляет перед названием сокращения в скобках? Примеры: (мл 01 ) Македонский Лев, (р ) Риганты. Том 1 и т.д. Как я понимаю эта настройка где-то в конвертере?

**Comment by petrovvlad** [comment]
Сокращения добавляются всегда или только при отправке по почте?

**Reply by NovomirK** [reply]
когда по почте пересылаю, при простой конвертации та же книга вот так: 001 Македонский Лев.mobi. Можетт метатеги где какие. Сейчас не могу посмотреть.

**Comment by petrovvlad** [comment]
Формат названия книги при конвертации можно задать во вкладке "Конвертация из fb2 d MOBI/AZW3/EPUB"

**Comment by petrovvlad** [comment]
Я не прав. Для того, что бы задать формат имени файла, нужно переключится на "отправить на устройство", задать формат, затем снова переключиться на "отправить на E-mail".
Неудобно. Но если это сработает то поправлю.

**Reply by NovomirK** [reply]
Да, работает. Спасибо большое!
То что в скобках это порядковый номер и сокращение названия серии, если переставить в конец, то очень удобно получается.

**Comment by petrovvlad** [comment]
Можете ещё проверить соберётся ли программа это командой:
`cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr .. && cmake --build .`

**Reply by NovomirK** [reply]
Да, собралась и установилась.

**Reply by petrovvlad** [reply]
Спасибо

**Reply by NovomirK** [reply]
Вам спасибо. Всегда рад помочь.
